File: PA8.java
// Minimum in a rotated sorted array 
class Solution {
    public int findMin(int[] nums) {
        int low =0 ;
        int high=nums.length-1;
        int mid=0;
        int min=nums[0];
        while(low<=high){
            mid= low + (high-low)/2;
            if(nums[mid]<min){
                min=nums[mid];
            }
            if(nums[mid]<nums[high]){
                high=mid-1;
            }
            else{
                low = mid+1;
            }
        }
        return min;
    }
}

File: PA_8.py
#Leetcode 153 : Find Minimum in Rotated Sorted Array
#Time Complexity : O(logn)
#Space Complexity : O(1)
class Solution(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0:
            return 0
        low = 0
        high = len(nums) - 1
        while low <= high:
            mid = (low + high)//2
            if low == high or (nums[mid] < nums[mid-1] and nums[mid] < nums[mid+1]):
                return nums[mid]
            elif nums[mid] < nums[high]:
                high = mid - 1
            else:
                low = mid + 1